src/main/java/appeng/core/sync/packets/PacketInventoryAction.java

```diff
@@ -79,54 +70,103 @@ public PacketInventoryAction( final InventoryAction action, final int slot, fina
 		this.slot = slot;
 		this.id = 0;
 		this.slotItem = slotItem;
+	}
 
-		final ByteBuf data = Unpooled.buffer();
-
-		data.writeInt( this.getPacketID() );
-		data.writeInt( action.ordinal() );
-		data.writeInt( slot );
-		data.writeLong( this.id );
+	// api
+	public PacketInventoryAction( final InventoryAction action, final int slot, final long id )
+	{
+		this.action = action;
+		this.slot = slot;
+		this.id = id;
+		this.slotItem = null;
+	}
 
-		if( slotItem == null )
+	@Override
+	public AppEngPacket onMessage( final PacketInventoryAction message, final MessageContext ctx )
+	{
+		if( ctx.side == Side.CLIENT )
 		{
-			data.writeBoolean( false );
+			return this.onMessageClientSide( message );
 		}
 		else
 		{
-			data.writeBoolean( true );
-			slotItem.writeToPacket( data );
+			return this.onMessageServerSide( message, ctx );
 		}
+	}
 
-		this.configureWrite( data );
+	@Override
+	public void fromBytes( final ByteBuf buf )
+	{
+		this.action = InventoryAction.values()[buf.readInt()];
+		this.slot = buf.readInt();
+		this.id = buf.readLong();
+		final boolean hasItem = buf.readBoolean();
+		if( hasItem )
+		{
+			try
+			{
+				this.slotItem = AEItemStack.loadItemStackFromPacket( buf );
+			}
+			catch( final IOException e )
+			{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)